### PR TITLE
Add date, time and number support

### PR DESCRIPTION
**Pull request by amcoder**
This PR adds support for date, times and numbers.

**Dates**
`{v, date}`
`{v, date, short}`
`{v, date, medium}`
`{v, date, long}`
`{v, date, full}` - Same as long

**Times**
`{v, time}`
`{v, time, short}`
`{v, time, medium}`
`{v, time, long}`
`{v, time, full}` - Same as long

**Numbers**
`{v, number}`

### CODE REVIEW

**Comment by tylermmorton** [comment]
This is really good. Since both `date` and `time` leverage the same formatter its going to be much easier to add skeleton support if we need it in the future. My only nit is to rename `date.go` to `datetime.go` 👍

**Comment by amcoder** [comment]
@Christian0411 @tylermmorton This is ready for review.

File: number.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package messageformat
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"golang.org/x/text/message"
```
Comment (tylermmorton) [inline number.go:8]: Awesome, I didn't know about this package.

File: formatter.go [hunk 2]
```diff
@@ -24,22 +24,7 @@ type FormatterOption func(f *formatter)
...
-		case language.German:
-			f.date = createGermanDateFormatter()
-			f.locale = locale
-		}
-	}
-}
-
-// TODO(tylermorton): replace this logic to use WithLocale
-// WithCulture applies the plurality culture to the formatter
-func WithCulture(culture string) FormatterOption {
```
Comment (tylermmorton) [inline formatter.go:40]: This is a good change. Thank you for figuring that out 👍

**Review by tylermmorton — APPROVED** [review]
LGTM!

**Review by Christian0411 — APPROVED** [review]
This looks good!

**Pull request merged**